File: HoYa/Entities/Inquiry.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace HoYa.Entities
{
    public class Inquiry : Detail<InquiryGeneral>//Process.Type.Value=="Inquiry"
    { 
        public virtual Guid? RecipeId { get; set; }
        [ForeignKey("RecipeId")]
        public virtual Recipe Recipe { get; set; }//原料耗材能源設備折舊成本
        public float? Price { get; set; }

        public string Delivery { get; set; }
        public string Remark { get; set; }
        public float? Quantity { get; set; }
    }

    public class InquiryGeneral : SimpleDetail<Process>
    {
        public virtual Guid? ContactId { get; set; }
        [ForeignKey("ContactId")]
        public virtual Contact Contact { get; set; }

        public virtual Guid? CustomerId { get; set; }
        [ForeignKey("CustomerId")]
        public virtual Person Customer { get; set; }
    }
}

File: HoYa.Web/src/entities/inquiry.ts
import { SimpleDetail,Detail } from "./entity";
import { Process } from "./workflow";
import { Recipe } from "./item";
import { Person, Contact } from "./person";
export class Inquiry extends Detail<InquiryGeneral>
{
    recipeId: string;
    recipe: Recipe;
    minAmount: number;
    quantity: number;
    price: number;

    delivery: string;
    remark: string;
    constructor(ownerId?:string) {
        super();
        this.ownerId = ownerId;
        this.quantity = 1;
    }
}

export class InquiryGeneral extends SimpleDetail<Process> {
    contactId: string;
    contact: Contact;
    customerId: string;
    customer: Person;
    constructor(ownerDefinitionBranchId?: string, ownerDefinitionChangeId?: string) {
        super();
        this.owner = new Process(ownerDefinitionBranchId, ownerDefinitionChangeId);
        this.contact = new Contact();
        this.contact.owner = new Person();
        this.customer = new Person();
    }
}